src/dfx/src/commands/deps/mod.rs

```diff
@@ -0,0 +1,43 @@
+use crate::lib::agent::create_agent_environment;
+use crate::lib::{environment::Environment, error::DfxResult};
+use crate::NetworkOpt;
+
+use clap::Parser;
+use tokio::runtime::Runtime;
+
+mod deploy;
+mod init;
+mod pull;
+
+/// Options for `dfx deps`.
+#[derive(Parser)]
+#[clap(name = "deps")]
+pub struct DepsOpts {
+    #[clap(flatten)]
+    network: NetworkOpt,
+
+    /// Arguments and flags for subcommands.
+    #[clap(subcommand)]
+    subcmd: SubCommand,
+}
+
+/// Subcommands of `dfx deps`
+#[derive(Parser)]
+enum SubCommand {
+    Pull(pull::DepsPullOpts),
+    Init(init::DepsInitOpts),
+    Deploy(deploy::DepsDeployOpts),
+}
+
+/// Executes `dfx deps` and its subcommands.
+pub fn exec(env: &dyn Environment, opts: DepsOpts) -> DfxResult {
+    let agent_env = create_agent_environment(env, opts.network.network)?;
```

Comment:
Recommend creating the anonymous agent environment here, since all of these commands should use the anonymous identity.  This will avoid prompting for credentials (for pem decryption) that aren't needed, as well as avoid holding onto an HSM session if so configured.